### PR TITLE
Added helper read-write option constructors + Error check

### PR DESCRIPTION
**Pull request by MehrdadMzfr**
Added few helper constructors:
-default
-infinite
-zero cache

+
added error check

Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Added few helper constructors:
-default
-infinite
-zero cache

### CODE REVIEW

File: AppCenterData/AppCenterData/MSReadOptions.h [hunk 1]
```diff
@@ -5,4 +5,27 @@
...
 @interface MSReadOptions : MSBaseOptions
 
+/**
+ * Initialize a MSReadOptions object.
+ *
+ * @param deviceTimeToLive time, in seconds, that document will be kept in the device cache. Time is releative to now.
+ *
+ * @return A MSWriteOptions instance.
+ */
+- (instancetype)initWithDeviceTimeToLive:(NSInteger)deviceTimeToLive;
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterData/MSReadOptions.h]: You don't need this here.

File: AppCenterData/AppCenterData/MSReadOptions.m [hunk 2]
```diff
@@ -1,8 +1,21 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#import "MSData.h"
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterData/MSReadOptions.m]: Please run formatter again.

File: AppCenterData/AppCenterData/MSReadOptions.m [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
 #import "MSReadOptions.h"
 
 @implementation MSReadOptions
 
+- (instancetype)initWithDeviceTimeToLive:(NSInteger)deviceTimeToLive {
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterData/MSReadOptions.m]: Remove this one. You don't need it.

File: AppCenterData/AppCenterData/MSWriteOptions.h [hunk 4]
```diff
@@ -5,4 +5,27 @@
...
 @interface MSWriteOptions : MSBaseOptions
 
+/**
+ * Initialize a MSWriteOptions object.
+ *
+ * @param deviceTimeToLive time, in seconds, that document will be kept in the device cache. Time is releative to now.
+ *
+ * @return A MSWriteOptions instance.
+ */
+- (instancetype)initWithDeviceTimeToLive:(NSInteger)deviceTimeToLive;
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterData/MSWriteOptions.h]: Same here.

File: AppCenterData/AppCenterData/MSWriteOptions.m [hunk 5]
```diff
@@ -1,8 +1,21 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#import "MSData.h"
 #import "MSWriteOptions.h"
 
 @implementation MSWriteOptions
 
+- (instancetype)initWithDeviceTimeToLive:(NSInteger)deviceTimeToLive {
```
Comment (jaelim-ms) [inline AppCenterData/AppCenterData/MSWriteOptions.m]: Remove this one.

File: AppCenterData/AppCenterData/MSBaseOptions.m [hunk 6]
```diff
@@ -2,13 +2,24 @@
 // Licensed under the MIT License.
 
 #import "MSBaseOptions.h"
+#import "AppCenter+Internal.h"
```
Comment (dhei) [inline AppCenterData/AppCenterData/MSBaseOptions.m]: nit: sort imports.

Comment (jaelim-ms) [reply AppCenterData/AppCenterData/MSBaseOptions.m]: This is correct.

File: AppCenterData/AppCenterData/MSReadOptions.m (hunk 2 above)
Comment (MehrdadMzfr) [reply AppCenterData/AppCenterData/MSReadOptions.m]: formatter doesn't sort it correctly. i re-ran and let it to  decide ;)

File: AppCenterData/AppCenterData/MSReadOptions.m [hunk 7]
```diff
@@ -2,7 +2,20 @@
...
 #import "MSReadOptions.h"
+#import "MSData.h"
 
 @implementation MSReadOptions
 
+- (instancetype)initWithDeviceTimeToLive:(NSInteger)deviceTimeToLive {
+  return [super initWithDeviceTimeToLive:deviceTimeToLive];
+}
+
++ (MSReadOptions *)createInfiniteCacheOptions {
```
Comment (clpolet) [inline AppCenterData/AppCenterData/MSReadOptions.m]: return `instancetype` instead.

File: AppCenterData/AppCenterData/MSReadOptions.m [hunk 8]
```diff
@@ -2,7 +2,20 @@
...
+  return [[MSReadOptions alloc] initWithDeviceTimeToLive:kMSDataTimeToLiveInfinite];
+}
+
++ (MSReadOptions *)createNoCacheOptions {
```
Comment (clpolet) [inline AppCenterData/AppCenterData/MSReadOptions.m]: return `instancetype` instead.

File: AppCenterData/AppCenterData/MSWriteOptions.h [hunk 9]
```diff
@@ -5,4 +5,18 @@
 
 @interface MSWriteOptions : MSBaseOptions
 
+/**
+ * Create an instance of MSWriteOptions with infinite cache time.
+ *
+ * @return A MSWriteOptions instance.
+ */
++ (MSWriteOptions *)createInfiniteCacheOptions;
```
Comment (clpolet) [inline AppCenterData/AppCenterData/MSWriteOptions.h]: return `instancetype` instead.

File: AppCenterData/AppCenterData/MSWriteOptions.h [hunk 10]
```diff
@@ -5,4 +5,18 @@
...
+
+/**
+ * Create an instance of MSWriteOptions with no cache.
+ *
+ * @return A MSWriteOptions instance.
+ */
++ (MSWriteOptions *)createNoCacheOptions;
```
Comment (clpolet) [inline AppCenterData/AppCenterData/MSWriteOptions.h]: return `instancetype` instead.

**Pull request merged**